ui_tests/pages/groups_page.py

```diff
@@ -0,0 +1,292 @@
+import time
+
+import pages.helper as helper
+from logger.logger import Logger
+from selenium.webdriver.common.by import By
+from webdriver.webriver_factory import WebDriverFactory
+
+logger = Logger().get_logger()
+driver = WebDriverFactory.get_driver()
+
+
+class GroupsPage:
+    def checkIfGroupsTableIsDisplayed(self):
+        logger.info("Check if groups page is displayed")
+        groups_table_xpath = "//div[@data-test='sc4snmp:group-table']"
+        groups_container = driver.find_element(By.XPATH, groups_table_xpath)
+        return groups_container.is_displayed()
+
+    def click_add_new_group_button(self):
+        logger.info(f"Click add new group button")
+        add_group_button_xpath = "//button[@data-test='sc4snmp:new-item-button']//span"
+        add_grp_btn = driver.find_element(By.XPATH, add_group_button_xpath)
+        add_grp_btn.click()
+        time.sleep(1)
+
+    def set_group_name(self, group_name):
+        logger.info(f"Set group name: {group_name}")
+        add_grp_input = self._get_group_name_input()
+        add_grp_input.send_keys(group_name)
+
+    def _get_group_name_input(self):
+        add_group_input_xpath = (
+            "//div[@data-test='sc4snmp:form:group-name-input']//span//input"
+        )
+        add_grp_input = driver.find_element(By.XPATH, add_group_input_xpath)
+        return add_grp_input
+
+    def click_submit_button_for_add_group(self):
+        logger.info(f"Click submit button")
+        add_group_button_xpath = (
+            "//button[@data-test='sc4snmp:form:submit-form-button']"
+        )
+        add_grp_btn = driver.find_element(By.XPATH, add_group_button_xpath)
+        add_grp_btn.click()
+        time.sleep(5)  # wait for group to be shown on the list
```

Comment:
there is no loading element, and this method is being used in multiple places 
- adding group
- editing group
it would be hard to add here explicit wait